pkg/api/v1/schema.go

```diff
@@ -121,30 +121,24 @@ type DiskDevice struct {
 }
 
 type DiskTarget struct {
-	// Device indicates the "logical" device name. The actual device name
-	// specified is not guaranteed to map to the device name in the guest OS. Treat
-	// it as a device ordering hint.
-	Device string `json:"dev,omitempty"`
+	// Bus indicates the type of disk device to emulate.
+	Bus string `json:"bus,omitempty"`
 	// ReadOnly
 	// Defaults to false
 	ReadOnly bool `json:"readonly,omitempty"`
 }
 
 type LunTarget struct {
-	// Device indicates the "logical" device name. The actual device name
-	// specified is not guaranteed to map to the device name in the guest OS. Treat
-	// it as a device ordering hint.
-	Device string `json:"dev,omitempty"`
+	// Bus indicates the type of disk device to emulate.
```

Comment:
Please enumerate all the bus types we support. Right now we don't have validation, but we will have it as soon as we have the validation webhook in place and we should tell the user what we will support. The comment here directly goes into the api reference. Does this one solely support scsi or also virtio?

Reply:
OK, will copy the content of https://github.com/libvirt/libvirt/blob/master/docs/schemas/domaincommon.rng#L1785 (libvirt schema) AND link it as reference.

Reply:
Let's start with adding those which make sense for us right now. The link just provides what is accepted in general, independent if we are talking about disks, luns, cdrom and floppy. I think the content of the link is just confusing. Directly stating what makes sense is more useful.